subprojects/precondition-tester/src/test/groovy/org/gradle/test/precondition/RemotePreconditionProbingTest.groovy

```diff
@@ -16,9 +16,8 @@
 
 package org.gradle.test.precondition
 
+import org.gradle.test.preconditions.UnitTestPreconditions
 
-import com.gradle.enterprise.testing.annotations.RemoteOnly
-
-@RemoteOnly
+@Requires(UnitTestPreconditions.OnRemoteTestDistributionExecutor)
```

Comment:
🤔 Interesting to have a `Requires` on a test that is supposed to check for allowed requires. I suppose this will work, though it feels a little bit strange. Can you add a comment on why we can't use `@RemoteOnly` here?

How much data would we lose if we only have one test class without the `@Requires` while keeping the new `OnRemoteTestDistributionExecutor` precondition?